src/cpp/include/openvino/genai/text2image/pipeline.hpp

```diff
@@ -0,0 +1,165 @@
+        }
+    };
+
+    explicit Text2ImagePipeline(const std::string& root_dir);
+
+    Text2ImagePipeline(const std::string& root_dir, const std::string& device, const ov::AnyMap& properties = {});
+
+    // creates either LCM or SD pipeline from building blocks
+    static Text2ImagePipeline stable_diffusion(
+        const std::shared_ptr<Scheduler>& scheduler_type,
+        const CLIPTextModel& clip_text_model,
+        const UNet2DConditionModel& unet,
+        const AutoencoderKL& vae_decoder);
+
+    // creates either LCM or SD pipeline from building blocks
+    static Text2ImagePipeline latent_consistency_model(
+        const std::shared_ptr<Scheduler>& scheduler_type,
+        const CLIPTextModel& clip_text_model,
+        const UNet2DConditionModel& unet,
+        const AutoencoderKL& vae_decoder);
+
+    GenerationConfig get_generation_config() const;
+    void set_generation_config(const GenerationConfig& generation_config);
+
+    // ability to override scheduler
+    void set_scheduler(std::shared_ptr<Scheduler> scheduler);
+
+    // with static shapes performance is better
+    void reshape(const int num_images_per_prompt, const int height, const int width, const float guidance_scale);
+
+    void compile(const std::string& device, const ov::AnyMap& properties = {});
+
+    // Returns a tensor with the following dimensions [num_images_per_prompt, height, width, 3]
+    ov::Tensor generate(const std::string& positive_prompt, const ov::AnyMap& properties = {});
+
+    template <typename... Properties>
+    ov::util::EnableIfAllStringAny<ov::Tensor, Properties...> generate(
+            const std::string& positive_prompt,
+            Properties&&... properties) {
+        return generate(positive_prompt, ov::AnyMap{std::forward<Properties>(properties)...});
+    }
+
+private:
+    class DiffusionPipeline;
+    std::shared_ptr<DiffusionPipeline> m_impl;
+
+    explicit Text2ImagePipeline(const std::shared_ptr<DiffusionPipeline>& impl);
+
+    class StableDiffusionPipeline;
+};
+
+//
+// Generation config properties
+//
+
+static constexpr ov::Property<std::string> prompt2{"prompt2"};
+static constexpr ov::Property<std::string> prompt3{"prompt3"};
+
+static constexpr ov::Property<std::string> negative_prompt{"negative_prompt"};
+static constexpr ov::Property<std::string> negative_prompt2{"negative_prompt2"};
+static constexpr ov::Property<std::string> negative_prompt3{"negative_prompt3"};
+
+static constexpr ov::Property<size_t> num_images_per_prompt{"num_images_per_prompt"};
+static constexpr ov::Property<float> guidance_scale{"guidance_scale"};
+static constexpr ov::Property<int64_t> height{"height"};
+static constexpr ov::Property<int64_t> width{"width"};
+static constexpr ov::Property<size_t> num_inference_steps{"num_inference_steps"};
+
+static constexpr ov::Property<std::shared_ptr<Generator>> random_generator{"random_generator"};
```

Comment:
There's missing `Scheduler`. Following `GenerationConfig`, there should be 3 places to set each of them: constructor, setter and `generate()`.

Reply:
in HF scheduler is not part of generate call / config.
It should be changed before generate / __call__ is called.